### PR TITLE
Us512

### PR DESCRIPTION
**Pull request by jmolive8**
- Made EMagnet slowly drain battery while active.
- Edited levels of EMagnet and their effects. Number of elements to upgrade may need work but I can easily fix that when we decide the total number of each element to have in the game.
- Magnetometer pickup now activates the first few Iron Veins and they should stay active when returning to the EMagnet level.
- Made EMagnet stop pulling towards Iron elements.

- [x] The code compiles
- [x] The code has been developer-tested
- [x] The script and each function has a description
- [x] The code follows guidelines listed in Quality Control Practices

### CODE REVIEW

**Comment by LopezRoman** [comment]
Play -> Tab -> Tab crashes game

image

File: Psyche/Assets/Scripts/Player/GammaView.cs
Comment (JoshBenn) [inline Psyche/Assets/Scripts/Player/GammaView.cs:1]: We need to identify what `GameStateManagerSn` is, already noted in a comment on slack.

File: Psyche/Assets/Scripts/Player/InventoryManager.cs
Comment (JoshBenn) [inline Psyche/Assets/Scripts/Player/InventoryManager.cs:1]: Already noted in a comment on slack --> no passing around strings if we can avoid it.

File: Psyche/Assets/Scripts/Player/IronVeinActivator.cs
Comment (JoshBenn) [inline Psyche/Assets/Scripts/Player/IronVeinActivator.cs:1]: Already noted --> no bracket-less blocks.

File: Psyche/Assets/Scripts/Player/PlayerDeath.cs
Comment (JoshBenn) [inline Psyche/Assets/Scripts/Player/PlayerDeath.cs:1]: Will be addressed with the style guide

**Review by JoshBenn — APPROVED** [review]
- [x] Comments and descriptions are comprehensible and add to the maintainability of the code
- [x] Comments are not too verbose or numerous
- [x] Code compiles
- [x] The goal of the US has been fulfilled
- [x] There are no merging conflicts
- [x] The code follows guidelines listed in Quality Control Practices

File: Psyche/Assets/Scripts/Player/GammaView.cs
Comment (LopezRoman) [reply Psyche/Assets/Scripts/Player/GammaView.cs:1]: GameStateManagerSn is the current scene being tracked by the Game State Manager.

**Pull request merged**